library/Graphite/Grapher.php

```diff
@@ -19,12 +19,18 @@ class Grapher extends GrapherHook
     protected $baseUrl = 'http://graphite.com/render/?';
     protected $serviceMacro = 'icinga2.$host.name$.services.$service.name$.$service.check_command$.perfdata.$metric$.value';
     protected $hostMacro = 'icinga2.$host.name$.host.$host.check_command$.perfdata.$metric$.value';
-    protected $imageUrlMacro = '&target=$target$&source=0&width=300&height=120&hideAxes=true&lineWidth=2&hideLegend=true&colorList=049BAF';
-    protected $largeImageUrlMacro = '&target=$target$&source=0&width=800&height=700&colorList=049BAF&lineMode=connected';
+    protected $imageUrlMacro = '&target=$target$&source=0&width=300&height=120&hideAxes=true&lineWidth=2&hideLegend=true&colorList=$colorList$&areaMode=$areaMode$&areaAlpha=$areaAlpha$';
+    protected $largeImageUrlMacro = '&target=$target$&source=0&width=800&height=700&colorList=$colorList$&lineMode=connected&areaMode=$areaMode$&areaAlpha=$areaAlpha$';
+    protected $DerivativeMacro = 'summarize(nonNegativeDerivative($target$),\'$summarizeInterval$\', \'$summarizeFunc$\')';
     protected $legacyMode = false;
     protected $graphiteKeys = array();
     protected $graphiteLabels = array();
     protected $areaMode = "all";
+    protected $graphType = "normal";
+    protected $summarizeInterval = "10min";
+    protected $summarizeFunc = "sum";
+    protected $areaAlpha = "0.1";
+    protected $colorList = "049BAF,EE1D00,04B06E,0446B0,871E10,CB315D,B06904,B0049C";
```

Comment:
Hi,
I agree that this would make the code more concise, but in our case, we find it useful to be able to overwrite these settings, case by case in Icinga's settings.
For example, the colorList is given by default to match the expected look-n-feel in Icinga's web pages. This can be set, as you say, in the template definitions. But for some graphs, we think useful to choose other colors. Network_IO check for instance is attributed blue and red colors (respectively for RX and TX).
This wouldn’t be possible if that settings is only set in the template definition.

What do you think?